pandas/core/accessor.py

```diff
@@ -291,8 +291,65 @@ def register_series_accessor(name):
     return _register_accessor(name, Series)
 
 
-@doc(_register_accessor, klass="Index")
```

Comment:
So any documentation changes should be made to the docstring of `_register_accessor` as it is shared with 3 other methods

Reply:
As I removed the decorator, the docstring is no longer shared

I believe it was shared earlier just to make the class visible in docs, but shared documentation wasn't right

Reply:
I think based on the issue suggestion:

> Add a link to an example or expand the topic to include a how to implement this. Also document what is required to make a custom indexer.

all 3 functions that share this docstring should be updated

Reply:
So shall I make appropriate docstring for other accessors as well?

Again the decorator will have to be removed as they won't share the docstrings anymore.

Reply:
I don't think the decorator needs to be removed. For example

> Also document what is required to make a custom indexer.

Should be the same and described for Series, DataFrame, and Index so the docstring should be shared

Reply:
> @mroeschke Some doc checks are failing, here is the issue... _register_accessor has it's own docs, which are appearing for all of 3 index, series and dataframe accessor. Now that I've put in the correct docs of this function.. the old docs are still being injected apart from the new docs.
> 
> How to remove this issue ? .. My guess is to remove the decorator itself for this `_register_index_accessor` method.
> 
> What should be the correct way to fix the error?

@mroeschke I've already tried that. CI fails .

Reply:
It detects dual docstring inserted, one due to decorator, and one, the modified string specific for register index accessor.

Reply:
Please note that, series, dataframe and index accessor work differently, so their docs can't be same, can they?